articles/introduction-to-kubernetes/index.md

```diff
@@ -0,0 +1,121 @@
+    app: api-server
+spec:
+  replicas: 10 # Defines the number of replicas in the deployment
+  selector:
+    matchLabels:
+      app: api-server
+  template: # Defines how a pod should look like
+    metadata:
+      labels:
+        app: api-server
+    spec:
+      containers: # Defines the containers that should be running in the pod
+      - name: rest-api-server
+        image: api-server
+        ports:
+        - containerPort: 80
+```
+### Service
+A service provides a static endpoint that can be used to direct traffic to the desired pod even if the pods change due to updating or scaling.
+
+The below YAML file shows how we can wrap a service around the deployment we created earlier.
+```yaml
+apiVersion: v1
+kind: service
+metadata:
+  name: rest-api-service
+  labels:
+    app: rest-api
+spec:
+  type: ClusterIP # Defines how we expose our endpoints
+  selector:
+    app: rest-api # Defines the pods to direct traffic to.
+  ports:
+  - protocol: TCP # Defines how clients communicate with our service.
+    port: 80
+
+```
+
+### Ingress
+Ingress makes it possible to expose our application running in a Kubernetes cluster to external traffic.
+Ingress makes it possible to define the services that should be public and the services that should remain private. The service for the user interface for our application can be public while the service running our backend application remains private.
+
+Below is an example of an Ingress configuration that exposes the user interface service.
+```yaml
+apiVersion: networking.k8s.io/v1beta1
+kind: Ingress
+metadata:
+  name: product-ingress
+  annotations:
+    kubernetes.io/ingress.class: "nginx"
+    nginx.ingress.kubernetes.io/rewrite-target: / # Defines the configuration for the ingress controller.
+    spec:
+    rules:
+    - http: # Defines how external traffic should access the service
+        paths:
+        - path: /app
+          backend: # Defines the service that the traffic is directed to.
+            serviceName: user-interface-service
+            servicePort: 80
+
+```
+
+### Other Kubernetes objects
+**Volumes** used to manage the file system within the pods. 
+**Secret** used for storing credentials i.e passwords.
+**ConfigMap** is used to specify application configurations that are mounted into a file.
+**HorizontalPodAutoscaler** is used to automatically scale pods based on resource utilization and availability.
+**StatefulSet** is similar to deployment but used to deploy stateful applications such as databases to the Kubernetes cluster.
+
+### Installing Kubernetes locally
+Kubernetes is bundled with [Docker desktop](https://www.docker.com/products/docker-desktop) which is available for both Windows and macOS. Tutorial on how to install docker on Linux can be found [here](https://kubernetes.io/docs/tasks/tools/install-kubectl/).
+
+To verify if your Kubernetes installation is working, run the below command.
+```bash
+$ kubectl version --client
+```
+
+### Conclusion
```

Comment:
Hey @paulodhiambo. Good job!
Just a few things to bring to your attention:
- Your article has a lot of passive voice statements which is not advisable when it comes to technical writing. Always try using active voice.
- You have quite a number of grammar mistakes. What you can do is before opening a PR you can use some tools such as Grammarly to cross-check your work.
You can access the links of various tools from Section's [resource](https://github.com/section-io/engineering-education/blob/master/new_contributors/resources-page.md) page.